app/models/stage.rb

```diff
@@ -252,19 +243,25 @@ def validate_not_auto_deploying_without_buddy
     end
   end
 
-  # has to be done after command_ids assignment is done
-  def append_new_command
-    return if @command.blank?
-    new_command = project.commands.new(command: @command)
-    previous = stage_commands.map(&:position).max || 0
-    stage_commands.build(command: new_command, position: previous + 1)
-  end
-
   def environment_lock?(lock)
     lock.resource_type == "Environment" && environments.any? { |e| lock.resource_equal?(e) }
   end
 
   def project_lock?(lock)
     lock.resource_type == "Project" && lock.resource_equal?(project)
   end
+
+  def find_or_create_stage_command(command_or_id, position)
+    command_or_id = command_or_id.to_s
+
+    if existing_command = command_or_id.match?(/\A\d+\z/) && Command.find_by(id: Integer(command_or_id))
+      if stage_command = stage_commands.detect { |sc| sc.command_id == existing_command.id }
+        stage_command
+      else
+        StageCommand.new(command: existing_command)
```

Comment:
did not assign project ?

Reply:
I think if the command already exists then it should either have the correct `project_id` already assigned to it, or it's a global command and so shouldn't have a `project_id` assigned. Is that right?